client/galaxy/scripts/mvc/workflow/workflow-manager.js

```diff
@@ -7,10 +7,49 @@ define(['mvc/workflow/workflow-connector'], function( Connector ) {
         this.name = null;
         this.has_changes = false;
         this.active_form_has_changes = false;
+        this.nodeLabels = {}; // TODO: track and enforce output labels also
     }
     $.extend( Workflow.prototype, {
-        create_node: function ( type, title_text, tool_id ) {
-            var node = this.app.prebuildNode( type, title_text, tool_id );
+        canLabelNodeWith: function( label ) {
+            if( label ) {
+                return ! (label in this.nodeLabels);
```

Comment:
I'll take an unobviously superior alternative if it feels more JavaScript-y, I don't do much JavaScript so I am happy to take feedback - even purely stylistic things.

For this particular set of functions and attribute I wanted a set which is only supported in EMCA 6, so I used the top voted pattern from [this StackOverflow article](http://stackoverflow.com/questions/7958292/mimicking-sets-in-javascript).